lib/eex/lib/eex/engine.ex

```diff
@@ -133,11 +144,19 @@ defmodule EEx.Engine do
     end
   end
 
+  def handle_expr(_buffer, marker, _expr) when marker in ["/", "|"] do
+    raise_marker_not_implemented(marker)
+  end
+
   def handle_expr(buffer, "", expr) do
     quote do
       tmp2 = unquote(buffer)
       unquote(expr)
       tmp2
     end
   end
+
+  defp raise_marker_not_implemented(marker) do
+    raise EEx.SyntaxError, message: "handle_expr/3 not implemented for <%#{marker} %>"
```

Comment:
Hrm, this message doesn't make much sense for users of the engine though. And those are the most likely to run into an issue. What about:

    raise EEx.SyntaxError, "unsupported EEx syntax <%#{marker} %> (the syntax is valid but not supported by the current EEx engine)"